org.eclipse.lsp4j.jsonrpc/src/main/java/org/eclipse/lsp4j/jsonrpc/RemoteEndpoint.java

```diff
@@ -221,24 +253,30 @@ protected boolean handleCancellation(NotificationMessage notificationMessage) {
 	protected void handleRequest(RequestMessage requestMessage) {
 		CompletableFuture<?> future;
 		try {
+			// Forward the request to the local endpoint
 			future = localEndpoint.request(requestMessage.getMethod(), requestMessage.getParams());
-		} catch (Throwable e) {
-			ResponseError errorObject = exceptionHandler.apply(e);
+		} catch (Throwable throwable) {
```

Comment:
Because we want to respond to the request in as many cases as possible. If an Error occurs (e.g. OutOfMemoryError), we send a response and then rethrow the error, which terminates the thread that is reading from the input stream.